### Scope
This notebook utilizes our Bayesian NMA implementation (MetaRank) for computing ATEs using the Senn 2013 dataset. 
<br>
We compare our results with R's `netmeta` package that implements a Frequentist Random Effects NMA model.

In [1]:
import pandas as pd

In [2]:
data = pd.read_csv('senn.csv')
data = data.rename({"treat1.long": "treatment", "treat2.long":"control"}, axis = 1)

In [3]:
data.head()

TE    seTE      treatment   control treat1 treat2            studlab
0 -1.90  0.1414      Metformin   Placebo   metf   plac       DeFronzo1995
1 -0.82  0.0992      Metformin   Placebo   metf   plac          Lewin2007
2 -0.20  0.3579      Metformin  Acarbose   metf   acar         Willms1999
3 -1.34  0.1435  Rosiglitazone   Placebo   rosi   plac       Davidson2007
4 -1.10  0.1141  Rosiglitazone   Placebo   rosi   plac  Wolffenbuttel1999

In [4]:
from nma import MetaRank
meta = MetaRank(baseline_tx='Placebo', 
                trace_samples=10000, 
                burn_ratio=0.2)

In [5]:
scores = meta.fit(data).predict()

Multiprocess sampling (4 chains in 4 jobs)
NUTS: [ate_std, tau, mur, mu0]


Sampling 4 chains for 10_000 tune and 10_000 draw iterations (40_000 + 40_000 draws total) took 375 seconds.
/Users/ag25710/miniconda3/envs/bcaus_nma/lib/python3.7/site-packages/arviz/data/io_pymc3.py:92: FutureWarning: Using `from_pymc3` without the model will be deprecated in a future release. Not using the model will return less accurate and less useful results. Make sure you use the model argument or call from_pymc3 within a model context.
  FutureWarning,


/Users/ag25710/miniconda3/envs/bcaus_nma/lib/python3.7/site-packages/arviz/data/io_pymc3.py:92: FutureWarning: Using `from_pymc3` without the model will be deprecated in a future release. Not using the model will return less accurate and less useful results. Make sure you use the model argument or call from_pymc3 within a model context.
  FutureWarning,


In [7]:
scores.sort_values('sucra', ascending=False)

treatment    ate  ate_lb  ate_ub     sucra
0  Rosiglitazone -1.236  -1.483  -0.979  0.893142
1      Metformin -1.130  -1.429  -0.838  0.783382
2   Pioglitazone -1.130  -1.556  -0.697  0.773188
3       Miglitol -0.949  -1.400  -0.504  0.610340
4       Acarbose -0.859  -1.298  -0.427  0.528358
5     Benfluorex -0.733  -1.292  -0.178  0.435337
6   Vildagliptin -0.696  -1.400   0.004  0.416500
7    Sitagliptin -0.570  -1.275   0.114  0.330059
8   Sulfonylurea -0.426  -0.893   0.041  0.214066
9        Placebo  0.000  -0.018   0.019  0.015628

### Comparison with R's NMA (frequentist)
https://rdrr.io/cran/netmeta/man/Senn2013.html
<br>
Random effects model
<br>
Treatment estimate (sm = '', comparison: other treatments vs 'Placebo'):

treatment | ATE | 95%-CI
-----|-----|----- 
Rosiglitazone| -1.2335| [-1.4839; -0.9830]|
Metformin|     -1.1268| [-1.4291; -0.8244]|
Pioglitazone|  -1.1291| [-1.5596; -0.6986]|
Miglitol|      -0.9497| [-1.4040; -0.4955]|
Acarbose|      -0.8418| [-1.3236; -0.3600]|
Benfluorex|    -0.7311| [-1.2918; -0.1705]|
Vildagliptin|  -0.7000| [-1.3927; -0.0073]|
Sitagliptin|   -0.5700| [-1.2640;  0.1240]|
Sulfonylurea|  -0.4166| [-0.8887;  0.0556]|
Placebo|             .|                  .|